Organisation editing

knirirr opened the pull request:
This PR offers the following:

- Logo upload when users create an organisation.
- Logo displayed on organisation pages. (*)
- Curators can edit organisations from their pages. (**)

(*) As best I can; someone else will have to change it if not to reviewers' tastes. 
(**) Except users and records, which are done separately, and parent/child organisations, which, if required, I would rather save for a separate ticket later. 

#1240

knirirr committed:
Allow adding logo when creating organisations. #1240

knirirr committed:
Autogenerated documentation files

knirirr committed:
Can now edit name, homepage, logo. #1240

knirirr committed:
Merge branch 'org_edit_1240' of github.com:FAIRsharing/fairsharing.github.io into org_edit_1240

knirirr committed:
Autogenerated documentation files

knirirr committed:
Edit organisation types. #1240

knirirr committed:
Edit ROR Link #1240.

knirirr committed:
Countries can be edited. #1240

knirirr committed:
Can edit alternative names #1240.

knirirr committed:
Added loader.

knirirr committed:
Show logo on edit form. #1240

knirirr committed:
Merge branch 'org_edit_1240' of github.com:FAIRsharing/fairsharing.github.io into org_edit_1240

knirirr committed:
Autogenerated documentation files

knirirr committed:
Removed console.log.

knirirr committed:
Merge branch 'org_edit_1240' of github.com:FAIRsharing/fairsharing.github.io into org_edit_1240

prakhyatox commented on src/components/Editor/Organisations/LinkOverlay.vue:
```diff
@@ -528,17 +533,29 @@
+              return;
+            }
+            _module.logoLoading = true;
+            let convertedFile = await toBase64(logo);
```
Conversion toBase64 could be called on click of ```Save new organisation``` button.

knirirr replied:
I've addressed this one here: https://github.com/FAIRsharing/fairsharing.github.io/pull/2264#discussion_r1521762363

prakhyatox commented on src/components/Editor/Organisations/LinkOverlay.vue:
```diff
@@ -528,17 +533,29 @@
-              };
-            });
-            reader.readAsDataURL(this.menus.newOrganisation.data.logo);
+        'menus.newOrganisation.data.logo': {
```
This watch could be removed.

knirirr replied:
On trying to remove this I see why it was originally done; if I attempt to access `this.menus.newOrganisation.data` within `createNewOrganisation` then it's always equal to `{}` (not so if it's done in the watcher).

prakhyatox commented on src/lib/Client/RESTClient.js:
```diff
@@ -605,6 +605,25 @@ class RESTClient {
     }
 
+    /**
+     * Create a given organisation
```
Should be ```Edit a given organisation```, since the function is about edit organisation

prakhyatox commented on src/views/Organisations/Organisation.vue:
```diff
@@ -183,21 +196,224 @@
         </v-fade-transition>
       </v-col>
     </div>
+    <!-- Edit existing organisation -->
```
If possible can Edit organisation be a separate component, giving the complexity and size of the file.

knirirr replied:
Having had  a look I don't think that this could easily be done.

prakhyatox replied:
Created ticket for the task: https://github.com/FAIRsharing/fairsharing.github.io/issues/2267

knirirr committed:
Fixed
https://github.com/FAIRsharing/fairsharing.github.io/pull/2264#discussion_r1521782503

knirirr committed:
Autogenerated documentation files

allysonlister commented:
Looking great. What are the conditions on the logo upload? I tried a jpg (rather large tbf) and it just stayed red with no message. A png was successful.

Here was the random image I unsuccessfully tried. It is completely unsuitable in size, which was one of the reasons I tried it. Just wondering about the reason it was rejected, and if we need to have some small text about the requirements for successful upload near or within the Logo upload text?
![RDA 10th anniversary Official Zoom Background](https://github.com/FAIRsharing/fairsharing.github.io/assets/143586/93d479a8-f4d7-44c0-99fd-dcfc001d7aa2)

knirirr commented:
The size limit is 1048576 bytes, i.e. ~1 MB.

knirirr committed:
Updated docs.

knirirr committed:
Merge branch 'dev' into org_edit_1240

knirirr committed:
Autogenerated documentation files

allysonlister commented:
> The size limit is 1048576 bytes, i.e. ~1 MB.

Thanks - is it worth putting this in the text somewhere? As it's only curators, it's not mandatory, but might be helpful to add any constraints to the text for forgetful people like me.

delphinedauga commented:
I've found this bug when I try to edit: the scrollbar on the side doesn't move the window up or down. I find this bug in Chrome and Opera. However, it works in Safari

https://github.com/FAIRsharing/fairsharing.github.io/assets/59688426/ead9b070-8f05-4f26-b467-6fa87d63a8fd

allysonlister commented:
FYI I did not have this bug in Firefox on Ubuntu.

knirirr commented:
I have also been unable to reproduce this.

knirirr committed:
Addded text detailing image requirements. #1240

knirirr committed:
Merge branch 'org_edit_1240' of github.com:FAIRsharing/fairsharing.github.io into org_edit_1240

allysonlister commented:
Actually, having just tried the scrolling again I'm getting the same behaviour on Firefox as @delphinedauga mentioned - I think I just didn't have my window small enough last time. If i make my window small enough that I would need to scroll, then no scrolling happens even though the little bar on the right moves up and down.

The message for the logo is great, thanks

knirirr committed:
Changed overlay to dialog. #1240

knirirr commented:
This is not really a bug; because I used an overlay scrolling will affect the element beneath which you won't see if the overlay is filling the screen. I've now changed it to a dialog which will change what is scrolled.
Resizing the browser window to something very small may still cause issues, but I was able to use this component successfully on  an emulated iPhone (both as overlay and dialog) so I think it will be OK.


![Screenshot from 2024-03-18 15-03-41](https://github.com/FAIRsharing/fairsharing.github.io/assets/3583375/9c745f5d-c8f2-4df0-aa11-276c0bbbeceb)

delphinedauga commented:
It works great! Perfect!

allysonlister reviewed (approved):
works great, thanks!
